source/spitebuf/src/lib.rs

```diff
@@ -178,6 +179,8 @@ pub fn cell_array<const N: usize, T: Sized>() -> [Cell<T>; N] {
 }
 
 impl<T> Cell<T> {
+    // TODO https://rust-lang.github.io/rust-clippy/master/index.html#/declare_interior_mutable_const
```

Comment:
That clippy lint is just flat-out wrong here; under the "known problems" section for that lint, the Clippy docs say:

> Known problems
>
> A “non-constant” const item is a legacy way to supply an initialized value to downstream static items (e.g., the `std::sync::ONCE_INIT` constant). In this case the use of const is legit, and this lint should be suppressed.

Throughout MnemOS, there are a bunch of other places where we allow this lint, sometimes with a comment that actually explains why, and sometimes with me just yelling at clippy to shut up: https://github.com/search?q=repo%3Atosc-rs%2Fmnemos%20declare_interior_mutable_const&type=code

We should probably change the comment here from a TODO to something like:

```suggestion
    // This constant is used in a static initializer, so the clippy lint does not apply.
    // See the "Known Problems" heading here:
    // https://rust-lang.github.io/rust-clippy/master/index.html#/declare_interior_mutable_const
```

Reply:
honestly, we should probably add a clippy config allowing this lint globally; i don't think it's ever actually helped catch a bug...